### DISCUSSION TITLE
Pseudo dice and training duration

### DISCUSSION THREAD

**Discussion by Peaceandmaths** [MIC-DKFZ/nnUNet#1935] [Q&A] [answered]
I have two questions : 

1) Is the pseudo-dice score in the progress.png file computed on the validation or training set? 
2) A colleague of mine trained nnunet on 100 images for 1000 epochs (similar images to mine, 3D CT/MR of the brain). The training took 20 hours in his case. Given that my dataset is 1400 images, I assumed it would take me around 10 times longer (~200 hours), but at the end, it took around 20h as well. I am surprised (in good). Why was it wrong to assume a linear relationship between number of images and duration ? What is the real bottleneck for training is it's not the amount of training images ? 

Thank you !

**Comment by aspil** [comment, marked as answer]
1. It is computed on the validation set.
2. Training time doesn't scale linearly with dataset size due to nnunet's patch-based training approach. Instead of going over the entire dataset explicitly during training, a patch-based approach is used where the model sees parts of images (patches) instead of the entire images.  Each epoch consists of a number of iterations, where each iteration processes a batch of these patches ([see code](https://github.com/MIC-DKFZ/nnUNet/blob/52cab354338ec32548b7b22600c94a6bf4cad1c2/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py#L1278)). The training time is independent of the dataset size because the total iterations are independent of the dataset size. Fabian explains the concept [here](https://github.com/MIC-DKFZ/nnUNet/issues/1292#issuecomment-1426218729).
